marklogic-client-api/src/main/java/com/marklogic/client/DatabaseClientFactory.java

```diff
@@ -1408,17 +1413,19 @@ static public DatabaseClient newClient(String host, int port, String basePath, S
 	  }
       services.connect(host, port, basePath, database, securityContext);
 
-      if (clientConfigurator != null) {
-          if (clientConfigurator instanceof OkHttpClientConfigurator) {
-              OkHttpClient okHttpClient = (OkHttpClient) services.getClientImplementation();
-              OkHttpClient.Builder clientBuilder = okHttpClient.newBuilder();
-              ((OkHttpClientConfigurator) clientConfigurator).configure(clientBuilder);
-              ((OkHttpServices) services).setClientImplementation(clientBuilder.build());
-          } else if (clientConfigurator instanceof HttpClientConfigurator) {
-              // do nothing as we no longer use HttpClient so there's nothing this can configure
-          } else {
-              throw new IllegalArgumentException("A ClientConfigurator must implement OkHttpClientConfigurator");
-          }
+      if (clientConfigurators != null) {
+		  clientConfigurators.forEach(configurator -> {
+			  if (configurator instanceof OkHttpClientConfigurator) {
+				  OkHttpClient okHttpClient = (OkHttpClient) services.getClientImplementation();
+				  OkHttpClient.Builder clientBuilder = okHttpClient.newBuilder();
+				  ((OkHttpClientConfigurator) configurator).configure(clientBuilder);
+				  ((OkHttpServices) services).setClientImplementation(clientBuilder.build());
+			  } else if (configurator instanceof HttpClientConfigurator) {
+				  // do nothing as we no longer use HttpClient so there's nothing this can configure
```

Comment:
Does this mean somebody might configure an HttpClientConfigurator object and expect it to do something, but then nothing happens?

Reply:
Correct, that's been the behavior since release 4.2.0. In fact, I'll submit a separate PR to mark that interface as deprecated so it can be removed in 7.0, along with this block of code.